script.js

```diff
@@ -1,47 +1,64 @@
-let books = [
-  {
-    title: 'book1',
-    author: 'author1',
-  },
-  {
-    title: 'book2',
-    author: 'author2',
-  },
-];
-
-function populateStorage(booksList) {
-  localStorage.setItem('books', JSON.stringify(booksList));
-}
+class Book {
+  constructor() {
+    if (localStorage.getItem('books')) {
+      this.list = JSON.parse(localStorage.getItem('books'));
+    } else { this.list = []; }
+  }
 
-function addBook(bookTitle, bookAuthor) {
-  const book = {
-    title: bookTitle,
-    author: bookAuthor,
-  };
-  books.push(book);
-  populateStorage(books);
-}
+  populateStorage() {
+    localStorage.setItem('books', JSON.stringify(this.list));
+  }
 
-function removeBook(index) {
-  books.splice(index, 1);
-  populateStorage(books);
+  add(bookTitle, bookAuthor) {
+    const book = {
+      title: bookTitle,
+      author: bookAuthor,
+    };
+    if (!this.search(bookTitle, bookAuthor)) {
+      this.list.push(book);
+      this.populateStorage();
+    } else {
+      const message = document.getElementById('error-message');
+      message.textContent = 'The book has already been added.';
+    }
+  }
+
+  remove(index) {
+    this.list.splice(index, 1);
+    this.populateStorage();
+  }
+
+  search(title, author) {
+    const book = this.list.filter((book) => book.title === title && book.author === author);
+    if (book.length > 0) return true;
+    return false;
+  }
+
+  refresh() {
+    if (localStorage.getItem('books')) {
+      this.list = JSON.parse(localStorage.getItem('books'));
+    }
+  }
+
+  getList() {
+    return this.list;
+  }
```

Comment:
- [ ] [OPTIONAL] You two have done great teamwork by refactoring using classes 👍🏽. However, I observed that after adding a new book, the input field still has the last added but as a placeholder, kindly fix this in order for us to have a good user experience.

Reply:
Thank you I've done it!